framework/src/modules/chain/init_steps/init_modules.js

```diff
@@ -19,29 +16,39 @@ const modulesList = {
 };
 
 module.exports = async scope => {
-	const tasks = {};
-
-	Object.keys(modulesList).forEach(name => {
-		tasks[name] = function(configModulesCb) {
-			const domain = require('domain').create();
+	const moduleNames = Object.keys(modulesList);
+	const modules = {};
+
+	const modulePromises = moduleNames.map(
+		moduleName =>
+			new Promise((resolve, reject) => {
+				const moduleDomain = domain.create();
+				const moduleCb = (err, object) => {
+					if (err) return reject(err);
+
+					return resolve(object);
+				};
+
+				moduleDomain.on('error', err => {
+					scope.components.logger.fatal(`Domain ${moduleName}`, {
+						message: err.message,
+						stack: err.stack,
+					});
+				});
 
-			domain.on('error', err => {
-				scope.components.logger.fatal(`Domain ${name}`, {
-					message: err.message,
-					stack: err.stack,
+				moduleDomain.run(() => {
+					scope.components.logger.debug('Loading module', moduleName);
+					// eslint-disable-next-line import/no-dynamic-require
+					const DynamicModule = require(modulesList[moduleName]);
+					return new DynamicModule(moduleCb, scope);
 				});
-			});
-
-			domain.run(() => {
-				scope.components.logger.debug('Loading module', name);
-				// eslint-disable-next-line import/no-dynamic-require
-				const DynamicModule = require(modulesList[name]);
-				return new DynamicModule(configModulesCb, scope);
-			});
-		};
+			})
+	);
+
+	(await Promise.all(modulePromises)).forEach((module, index) => {
+		modules[moduleNames[index]] = module;
```

Comment:
instead of declearing at top, we should map it to the new variable.
```
const resolvedModules = await Promise.all(modulePromises);
const modules = resolvedModules.reduce((prev, module, index) => ({ ...prev, [index]: module }), {});
```